upstairs/src/main.rs

```diff
@@ -70,13 +74,47 @@ mod tests {
             SocketAddrV4::new(Ipv4Addr::new(192, 168, 1, 2), 3801)
         );
     }
+
+    #[test]
+    fn test_key() {
+        let key_bytes =
+            base64::decode("9YGqFSwBHCX/IbjstbI1WuUPKOfwrwNAJSFzUN2w4iU=")
+                .unwrap();
+
+        let opt = Opt::from_string(
+            "-- -t 192.168.1.1:3801 -k 9YGqFSwBHCX/IbjstbI1WuUPKOfwrwNAJSFzUN2w4iU=".to_string(),
+        )
+        .unwrap();
+
+        if let Some(key) = &opt.key {
+            assert_eq!(
+                base64::decode(key).expect("base64 decode failed"),
+                key_bytes
+            );
+        } else {
+            panic!("bad news");
```

Comment:
maybe panic with `failed to encode key option` or some more info on why the news is bad :)

Reply:
:+1: